controllers/hvpa_controller.go

```diff
@@ -1123,8 +1126,13 @@ func (r *HvpaReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 			hvpa.Status.HvpaStatus.LastScaleType.Vertical = autoscalingv1alpha1.Up
 		}*/
 	}
+
+	if r.initScalingMetrics(hvpa, obj) != nil {
```

Comment:
This will initialise all the metrics at every reconcile independent of whether any scaling is done or not. we should initialise only if not already initialised, right? Otherwise, when there is no scaling required, previous scaling data will also be lost. WDYT?

Reply:
I could not find any prometheus API to check if a metric is initialised. Hence, this suboptimal solution.